### PR TITLE
(chore): use expected errors in 29fee

### PR DESCRIPTION
**Pull request by bznein**
## Description

closes: #7190

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/29-fee/ibc_middleware_test.go [hunk 1]
```diff
@@ -33,44 +36,44 @@ func (suite *FeeTestSuite) TestOnChanOpenInit() {
...
 			true,
-			true,
+			nil,
 		},
 		{
 			"success - fee version not included, only perform mock logic",
 			ibcmock.Version,
-			true,
 			false,
+			nil,
```
Comment (bznein) [inline modules/apps/29-fee/ibc_middleware_test.go:52]: Moved this up as it's a success case

File: modules/apps/29-fee/ibc_middleware_test.go [hunk 2]
```diff
@@ -135,7 +138,7 @@ func (suite *FeeTestSuite) TestOnChanOpenInit() {
 
 					suite.Require().NoError(err, "unexpected error from version: %s", tc.version)
 				} else {
-					suite.Require().Error(err, "error not returned for version: %s", tc.version)
+					suite.Require().True(errors.Is(err, tc.expErr) || strings.Contains(err.Error(), tc.expErr.Error()), "error no returned for version: %s or error %s is not %s", tc.version, err, tc.expErr)
```
Comment (bznein) [inline modules/apps/29-fee/ibc_middleware_test.go:141]: This line will come up several times in the PR. It's ugly, but the issue is that some of the errors that we get (especially from `sdk`) are not wrapped, so we can't check for a specific error but rather we have to rely on the error string. It's not ideal, but the other option would be to catch every usage of those errors and wrap them ourselves with something else

File: modules/apps/29-fee/ibc_middleware_test.go [hunk 3]
```diff
@@ -325,19 +328,16 @@ func (suite *FeeTestSuite) TestOnChanCloseInit() {
...
 		},
 		{
-			"application callback fails", func() {
-				suite.chainA.GetSimApp().FeeMockModule.IBCApp.OnChanCloseInit = func(
-					ctx sdk.Context, portID, channelID string,
-				) error {
-					return fmt.Errorf("application callback fails")
-				}
-			}, false,
+			"fee module is not enabled", func() {
```
Comment (bznein) [inline modules/apps/29-fee/ibc_middleware_test.go:337]: I moved stuff around here to have success cases before failures

File: modules/apps/29-fee/keeper/grpc_query_test.go [hunk 4]
```diff
@@ -24,7 +24,7 @@ func (suite *KeeperTestSuite) TestQueryIncentivizedPackets() {
 	testCases := []struct {
 		name     string
 		malleate func()
-		expPass  bool
+		errMsg   string
```
Comment (bznein) [inline modules/apps/29-fee/keeper/grpc_query_test.go:27]: This file relies entirely on error messages as all the errors we get from GRPC queries are not wrapped

File: modules/apps/29-fee/keeper/grpc_query_test.go [hunk 5]
```diff
@@ -643,16 +643,61 @@ func (suite *KeeperTestSuite) TestQueryCounterpartyPayee() {
...
 				suite.Require().Equal(expCounterpartyPayeeAddr.String(), res.CounterpartyPayee)
 			} else {
-				suite.Require().Error(err)
+				suite.Require().ErrorContains(err, tc.errMsg)
 			}
 		})
 	}
 }
 
+func (suite *KeeperTestSuite) TestQueryFeeEnabledChannelsWithPagination() {
```
Comment (bznein) [inline modules/apps/29-fee/keeper/grpc_query_test.go:656]: This was really interesting. 

In this test, that was before a test case of another one, we create 10 extra channel, then get the first 5 from pagination and compare them.  However, when building them, we loop over an integer and create them. 

BUT, the logic that handles pagination paginates them based on their *lexicographical order*.
This means that if our channels are all after 10 (and before 100, so as to have 2 digits), the two orders are equal.

But in our case, depending on how many tests were ran before that, we could have the case in which our channels were 1,2,3,4,5,6,7,8,9,10, but the pagination would return 1, *10*, 3,4,5,6,7,8,9, making the test fail!

Thanks @chatton for helping me debug this!

File: modules/apps/29-fee/keeper/msg_server_test.go [hunk 6]
```diff
@@ -21,90 +25,92 @@ func (suite *KeeperTestSuite) TestRegisterPayee() {
...
+			ibcerrors.ErrUnauthorized,
 		},
 	}
 
 	for _, tc := range testCases {
 		tc := tc
 
-		suite.SetupTest()
-		suite.path.Setup()
+		suite.Run(tc.name, func() {
```
Comment (bznein) [inline modules/apps/29-fee/keeper/msg_server_test.go:69]: There are a few instances in this PR where I introduced `t.Run` or `suite.Run`. It helped me debug things but I also think it's better to have it this way, even though it increases the diff. Happy to revert if people find it would be better suited for another PR

File: modules/apps/29-fee/types/fee_test.go [hunk 7]
```diff
@@ -198,31 +201,33 @@ func TestPacketFeeValidation(t *testing.T) {
...
-		if tc.expPass {
-			require.NoError(t, err, tc.name)
-		} else {
-			require.Error(t, err, tc.name)
-		}
+			if tc.expErr == nil {
+				require.NoError(t, err, tc.name)
+			} else {
+				require.True(t, errors.Is(err, tc.expErr) || strings.Contains(err.Error(), tc.expErr.Error()), err.Error())
+			}
```
Comment (chatton) [inline modules/apps/29-fee/types/fee_test.go:230]: I wonder would it be worth while turning this into some helper within the tests, seems like we need it in a good few places where errors aren't wrapped.

**Review by chatton — APPROVED** [review]
LGTM! Thanks for going through all these, it ended up being quite a bit :1st_place_medal: 

Only suggestion is maybe adding a helper for that error checking condition, otherwise looks great!

File: modules/apps/29-fee/types/fee_test.go (hunk 7 above)
Comment (bznein) [reply modules/apps/29-fee/types/fee_test.go:230]: yeah makes sense, done!

File: modules/apps/29-fee/ibc_middleware_test.go [hunk 8]
```diff
@@ -135,7 +137,7 @@ func (suite *FeeTestSuite) TestOnChanOpenInit() {
 
 					suite.Require().NoError(err, "unexpected error from version: %s", tc.version)
 				} else {
-					suite.Require().Error(err, "error not returned for version: %s", tc.version)
+					ibctesting.RequireErrorIsOrContains(suite.T(), err, tc.expErr, "error no returned for version: %s or error %s is not %s", tc.version, err, tc.expErr)
```
Comment (crodriguezvega) [inline modules/apps/29-fee/ibc_middleware_test.go:140]: You forgot maybe to push this new function?

Comment (crodriguezvega) [reply modules/apps/29-fee/ibc_middleware_test.go:140]:
```suggestion
					ibctesting.RequireErrorIsOrContains(suite.T(), err, tc.expErr, "error not returned for version: %s or error %s is not %s", tc.version, err, tc.expErr)
```

File: modules/apps/29-fee/keeper/grpc_query_test.go (hunk 5 above)
Comment (crodriguezvega) [reply modules/apps/29-fee/keeper/grpc_query_test.go:656]: Very interesting! And thank you both for debugging and fixing it!

**Review by crodriguezvega — COMMENTED** [review]
Looks like we are missing the function `RequireErrorIsOrContains`. Other than that it looks good; thanks again for continuing with these refactors!

**Review by crodriguezvega — APPROVED** [review]
🎉
